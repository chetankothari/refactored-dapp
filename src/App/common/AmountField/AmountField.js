import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AmountField extends Component {
  updateAmount = (event) => {
    const { value: amount } = event.target;
    this.props.amountChangeHandler(amount);
  };

  render() {
    const { id, amount, placeholder } = this.props;
    return (
      <input
        type="number"
        name={`${id}-amount`}
        value={amount}
        placeholder={placeholder}
        onChange={this.updateAmount}
      />
    );
  }
}

AmountField.propTypes = {
  amount: PropTypes.number,
  amountChangeHandler: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
};

AmountField.defaultProps = {
  placeholder: '',
};

export default AmountField;
