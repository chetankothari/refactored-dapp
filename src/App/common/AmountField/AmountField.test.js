import React from 'react';
import { shallow } from 'enzyme';

import AmountField from "./AmountField";

describe('AmountField', function() {
  it('renders input field with default props', () => {
    const updateAmount = jest.fn();
    const wrapper = shallow(<AmountField amountChangeHandler={updateAmount} id="unique-field-id" />);
    const inputField = wrapper.find('input').at(0);

    expect(inputField).toMatchSnapshot();
  });

  it('renders input field with overridden defaults', () => {
    const updateAmount = jest.fn();
    const wrapper = shallow(
      <AmountField
        amountChangeHandler={updateAmount}
        id="unique-field-id"
        placeholder="overridden-placeholder"
        amount={1}
      />
    );
    const inputField = wrapper.find('input').at(0);

    expect(inputField).toMatchSnapshot();
  });

  it('triggers amountChangeHandler on updating the input field', () => {
    const updateAmount = jest.fn();
    const wrapper = shallow(<AmountField amountChangeHandler={updateAmount} id="unique-field-id" />);
    const inputField = wrapper.find('input').at(0);

    inputField.simulate('change', { target: { value: 1 } });

    expect(updateAmount).toHaveBeenCalledWith(1);
  });

  it('triggers amountChangeHandler on updating the input field with existing value', () => {
    const updateAmount = jest.fn();
    const wrapper = shallow(<AmountField amountChangeHandler={updateAmount} id="unique-field-id" amount={0.001} />);
    const inputField = wrapper.find('input').at(0);

    inputField.simulate('change', { target: { value: 0.0010 } });

    expect(updateAmount).toHaveBeenCalledWith(0.0010);
  });
});
