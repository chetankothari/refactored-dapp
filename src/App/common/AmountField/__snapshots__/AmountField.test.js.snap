// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountField renders input field with default props 1`] = `
<input
  name="unique-field-id-amount"
  onChange={[Function]}
  placeholder=""
  type="number"
/>
`;

exports[`AmountField renders input field with overridden defaults 1`] = `
<input
  name="unique-field-id-amount"
  onChange={[Function]}
  placeholder="overridden-placeholder"
  type="number"
  value={1}
/>
`;
